.github/workflows/pr.yml

```diff
@@ -43,7 +47,9 @@ jobs:
     - name: Generate JUnitReport
       if: always()
       run: ant -f tlatools/org.lamport.tlatools/customBuild.xml test-report
-    - name: Upload raw unit test results
+    - name: Run unit tests on dist and generate coverage report
+      run: ant -f tlatools/org.lamport.tlatools/customBuild.xml test-dist -Dtest.halt=true code-coverage-report
```

Comment:
How long does this take to run? Ideally we want the CI to complete within ten minutes.

Reply:
> How long does this take to run? Ideally we want the CI to complete within ten minutes.

Where is this number coming from?

Reply:
That's how fast the CI was running before two weeks ago. I put in quite a bit of effort to get it there!

Reply:
OK, so "we" was used in the sense of the royal "we."

Reply:
So do you care about how fast the CI runs or not?

Reply:
> Edit: I see from the CI run that this takes as long as the original unit tests. I don't think we should run the unit tests twice in a row, which doubles the total CI runtime.

I think there are two improvements that can be done here:
1. Run test-dist using threads. I just tried a test run, it succeeds in less time and the coverage is same
2. Steps cannot run in parallel, but I could create a new job to run test-dist and upload report + publish comment. This should help further reduce the total ci run time.

How does these sound?

Reply:
Why does it have to run twice? Is it just because of that one problematic test case? We could pull that out into its own `junit` task invocation where we don't track code coverage, right?

Reply:
Yes, that's an option but I think the priority is on keeping the customBuild file as simple as possible

Reply:
I guess you can count me as advocating for adding 50 more lines of XML instead of ensuring all future CI runs take twice as long while adding no additional test value (how much extra CO2 will that produce?) - we also already have multiple `junit` task invocations in the `test` target for the non-parallel tests.

Reply:
As an alternative, we could skip this work altogether. I think keeping an eye on coverage is good but it's not really vital as long as new contributions include tests. The code-coverage-report ant target is already useful by itself, especially if anyone wants to start contributing some tests. I'll resolve this conversation for now. Let's keep discussing in the [ticket](https://github.com/tlaplus/tlaplus/issues/981) to avoid losing good comments in PRs. Thanks for the feedback!